provider/workspace_preset.go

```diff
@@ -0,0 +1,39 @@
+package provider
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type WorkspacePreset struct {
+	Name       string            `mapstructure:"name"`
+	Parameters map[string]string `mapstructure:"parameters"`
+}
+
+func workspacePresetDataSource() *schema.Resource {
+	return &schema.Resource{
+		SchemaVersion: 1,
+
+		Description: "",
+		ReadContext: func(ctx context.Context, rd *schema.ResourceData, i interface{}) diag.Diagnostics {
+			rd.SetId(uuid.NewString())
+
+			return nil
```

Comment:
TODO: 
double check whether I'm missing anything here. 
If not, comment why its implementation is this bare.
Also check the purpose and correct value for `rd.SetId(...)`